plugins/postgres/dbt/adapters/postgres/connections.py

```diff
@@ -22,6 +22,7 @@ class PostgresCredentials(Credentials):
     user: str
     port: Port
     password: str  # on postgres the password is mandatory
+    connect_retries: int = 3
```

Comment:
Yes, you are correct.

This idea was taken from dbt-snowflake which already implements it's own version of retrying. That being said, I'm not a fan of the confusion between retry options and connection options: for example, `connect_timeout` is a Postgres connection option, yet in dbt-snowflake it's a retry option, which is why here I'm using `retry_timeout` instead, but this means different adapters have different configuration names for what it's essentially the same thing.

Reply:
@jtcohen6 do you have any opinions here?

Reply:
Personally, I'm a fan of extensive configurations with sensible defaults, but defining said defaults can be hard, and in the mean-time the user experience would suffer. So, I do see the argument for simplicity, and I thank you for your insight.

The question that is still open for me is where would the `number_of_retries` configuration option be exposed. I think there are 3 options:
1. As a user configuration (similar to `fail_fast`, `use_colors`, or other [flags](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/flags.py)).
2. As a top level profile option (similar to `target`).
3. In each individual target, which means in each adapters credentials (next to other connection options like `password`, that's how it's done in this PR currently).

Given that each target of a given adapter will have the same retrying behavior, I think the 3rd option would be missed a lot or would lead to a lot of duplication.

I haven't see top level profile options used for anything but the default target and outputs, so I'd think 2nd option would also be out.

And for these reasons I'm leaning towards 1. But do you have any thoughts on this?

Reply:
Yeah, `retries` definitely belongs in with other connection options, I do agree with that. From a user perspective, I still think it would be nice to be able to set a global `retries`, but that can perhaps be left for the future as I do agree that it doesn't go well with other CLI flags.

I'll try this out in a follow-up commit. Thanks :+1: